File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    const hallOfFame = [];
    const answer = [];

    for (let i = 0; i < score.length; i++) {
        hallOfFame.push(score[i]);                    // 오늘 점수 추가
        hallOfFame.sort((a, b) => b - a);             // 내림차순 정렬 (높은 점수 먼저)

        if (hallOfFame.length > k) {
            hallOfFame.pop();                         // k명 초과 시 가장 낮은 점수 제거
        }

        answer.push(hallOfFame[hallOfFame.length - 1]); // 현재 명예의 전당 최하위 점수 추가
    }

    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    hall_of_fame = []
    answer = []

    for s in score:
        hall_of_fame.append(s)
        hall_of_fame.sort(reverse=True)  # 내림차순 정렬
        if len(hall_of_fame) > k:
            hall_of_fame.pop()  # k개 넘으면 가장 낮은 점수 제거
        answer.append(hall_of_fame[-1])  # 현재 명예의 전당의 최하위 점수 기록

    return answer
